synapse/handlers/__init__.py

```diff
@@ -54,7 +55,12 @@ def __init__(self, hs):
         self.directory_handler = DirectoryHandler(hs)
         self.typing_notification_handler = TypingNotificationHandler(hs)
         self.admin_handler = AdminHandler(hs)
+        asapi = ApplicationServiceApi(hs)
         self.appservice_handler = ApplicationServicesHandler(
-            hs, ApplicationServiceApi(hs)
+            hs, asapi, AppServiceScheduler(
+                clock=hs.get_clock(),
+                store=hs.get_datastore(),
+                as_api=asapi
+            )
```

Comment:
Oh, right, for tests? It just seemed a bit odd not to either put it in the `HomeServer` object or move it into the constructor of `ApplicationServicesHandler`.